db/migrate/20171026114327_add_deleted_on_to_container_quota_and_items.rb

```diff
@@ -9,6 +9,8 @@ def change
     add_index :container_quotas, :deleted_on
 
     add_timestamps :container_quota_items, :null => true # temporarily for backfilling, then disallow nulls
+    ContainerQuotaItem.reset_column_information
```

Comment:
I'm surprised you put this in the middle, from your description of the problem I'd expect clearing things "between" the migrations.
BTW, down direction runs `change` statements in reverse — did you test down also works?  (it's likely caching is not a problem if this migration runs first anyway.)

Hmm, I guess the way `change` works, to implement running in reverse, is first only recording steps to be done.  Since this line is not in a `reversible` block, it probably runs before all the steps, for both directions, whereever you place it?

BTW, why didn't the migrate-all-the-way-up test catch this?

[I don't care enough to play with this myself, and not saying we should touch it again, if it works]